ground/src/main/java/com/google/android/ground/ui/home/SignOutConfirmationDialog.kt

```diff
@@ -15,41 +15,87 @@
  */
 package com.google.android.ground.ui.home
 
+import android.content.Context
 import androidx.compose.material3.AlertDialog
 import androidx.compose.material3.Button
+import androidx.compose.material3.OutlinedButton
 import androidx.compose.material3.Text
 import androidx.compose.material3.TextButton
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.MutableState
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.res.stringResource
 import com.google.android.ground.R
+import com.google.android.material.color.MaterialColors
 
 @Composable
-fun SignOutConfirmationDialog(openDialog: MutableState<Boolean>, signOutCallback: () -> Unit) {
+fun SignOutConfirmationDialog(
```

Comment:
Sorry just noticing this now; this dialog class now represents two dialogs, not just the Sign Out confirmation (see [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle)). Can you instead create a separate class called "UserDetailsDialog" and move the new dialog there?

Reply:
Sure!